### PR TITLE
chore(examples): Convert `blog` example to TypeScript

### PR DESCRIPTION
**Pull request by maxproske**
Converted Blog example over to TypeScript to match the Contribution guidelines.

I also simplified the example by removing `_document.js` and prettier from `package.json`.

## Documentation / Examples

- [X] Make sure the linting passes by running `pnpm lint`
- [X] The examples guidelines are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing.md#adding-examples)

### CODE REVIEW

**Comment by maxproske** [comment]
This example is built with a static site generator, [Nextra](https://github.com/shuding/nextra), and it could be renamed accordingly.

Should I move this example from `blog` to `blog-nextra`, and put a redirect notice at `examples/blog/README.md`?

File: examples/blog/package.json [hunk 1]
```diff
@@ -6,19 +6,18 @@
...
   },
   "dependencies": {
-    "gray-matter": "^4.0.2",
+    "gray-matter": "^4.0.3",
     "next": "latest",
-    "nextra": "^2.0.0-beta.5",
-    "nextra-theme-blog": "^0.1.5",
-    "react": "^17.0.2",
-    "react-dom": "^17.0.2",
+    "nextra": "^1.1.0",
```
Comment (balazsorban44) [inline examples/blog/package.json:11]: Why were we downgrading here? This was bumped here: https://github.com/vercel/next.js/pull/32256


@shu do you have a preference?

**Review by balazsorban44 — COMMENTED** [review]
Thanks, added a small note

File: examples/blog/package.json (hunk 1 above)
Comment (maxproske) [reply examples/blog/package.json]: @balazsorban44 I reverted nextra back to the latest beta release.

**Comment by maxproske** [comment]
@balazsorban44 Made all your requested changes - Ready for review!

**Review by balazsorban44 — APPROVED** [review]
Thanks!

**Pull request merged**